openprescribing/dmd2/forms.py

```diff
@@ -0,0 +1,56 @@
+from django import forms
+
+from crispy_forms.helper import FormHelper
+from crispy_forms.layout import Layout, Field, ButtonHolder, Submit
+from crispy_forms.bootstrap import InlineCheckboxes
+
+
+obj_types_choices = [
+    ('vtm', 'VTMs'),
+    ('vmp', 'VMPs'),
+    ('amp', 'AMPs'),
+    ('vmpp', 'VMPPs'),
+    ('ampp', 'AMPPs'),
+]
+
+include_choices = [
+    ('unavailable', 'Unavailable items'),
+    ('invalid', 'Invalid items'),
+    ('no_bnf_code', 'Items with no BNF code'),
+]
+
+class SearchForm(forms.Form):
+    q = forms.CharField(
+        label='Query or SNOMED code',
+        min_length=3,
+    )
+    obj_types = forms.MultipleChoiceField(
+        label='Search...',
+        choices=obj_types_choices,
+        required=False,
+        initial=[tpl[0] for tpl in obj_types_choices],
+    )
+    include = forms.MultipleChoiceField(
+        label='Include...',
+        choices=include_choices,
+        required=False,
+    )
```

Comment:
I feel these fields need a bit of explanation, though I can see that's hard to do automagically with crispy forms.  Perhaps with a side bar or an asterisked bit of text?

In particular, we need to explain that "not available" [excludes](https://github.com/ebmdatalab/openprescribing/blob/7f30479e92a9b8039b941c81d85c583444a2db9f/openprescribing/dmd2/managers.py) items marked in dm+d as "Actual Products Not Available" and "invalid" means... um... I'm not sure...

Looking at the [spec](https://www.nhsbsa.nhs.uk/sites/default/files/2017-02/Technical_Specification_of_data_files_R2_v3.1_May_2015.pdf) I think `invalid` is reserved for things where there have been mistakes, or similar. Perhaps we should never show these things. For example, p43 of the [implementation guide](https://www.nhsbsa.nhs.uk/sites/default/files/2017-02/dmd_Implemention_Guide_%28Primary_Care%29_v1.0.pdf) says you should not include invalid items in the DMDProducts table

Or perhaps we just need an explanation "`invalid` is anything marked as such in the dm+d, such as errors that have been corrected"

Reply:
> "invalid" means... um... I'm not sure...

The tech spec just says "If set to 1 indicates this is an invalid entry in file."  It should be clear to a switched-on user that this is filtering by a value in our database, so I don't actually think it needs an explanation -- we don't provide explanations of other fields.

As for unavailable items, what about "Unavailable items are not available to be prescribed and/or have been discontinued", which covers the different ways that an object can be unavailable.

Reply:
I'm still not convinced about `invalid`, but I guess we can wait to see if we get any queries and deal with it then...